### PR TITLE
feat: add support for commit revert API (#991)

### LINKED ISSUE #991: How to revert a commit of a project
## Description of the problem, including code/CLI snippet
I need to remove an specific commit in a project

## Expected Behavior
https://docs.gitlab.com/ce/api/commits.html#revert-a-commit

## Specifications

  - python-gitlab version: 
  - API version you are using (v3/v4): v4
  - Gitlab server version (or gitlab.com): 12.4

### PR DESCRIPTION
**Pull request by nejch**
Closes #991.

### CODE REVIEW

**Comment by nejch** [comment]
Hmm, Travis was failing on something seemingly unrelated in functional tests (https://travis-ci.org/python-gitlab/python-gitlab/jobs/652064210), so I re-opened the PR to trigger it again and now it's happy. ¯\\\_(ツ)\_/¯

**Comment by nejch** [comment]
I was wondering if maybe `revert()` should return a commit object instead of a dict. It seems like most other update methods also just return the server response but since this creates a new resource, people might want to manipulate that new commit directly - I just didn't see any other examples (`cherry_pick` doesn't return anything atm).

File: gitlab/tests/test_gitlab.py [hunk 1]
```diff
@@ -794,6 +794,58 @@ def resp_deactivate(url, request):
...
+            content = """{
+            "id": "8b090c1b79a14f2bd9e8a738f717824ff53aebad",
+            "short_id": "8b090c1b",
+            "title":"Revert \\"Initial commit\\""
+            }"""
+            content = content.encode("utf-8")
+            return response(200, content, headers, None, 5, request)
+
+        with HTTMock(resp_get_project, resp_get_commit):
+            project = self.gl.projects.get(1)
```
Comment (max-wittig) [inline gitlab/tests/test_gitlab.py]: suggestion: Make the tests more compact by using lazy=True

```python
with HTTMock(resp_get_commit):
    project = self.gl.projects.get(1, lazy=True)
```

Comment (nejch) [reply gitlab/tests/test_gitlab.py]: Done :) thanks

**Comment by max-wittig** [comment]
Thanks for the MR!

**Pull request merged**